### PR TITLE
Add GDI Covert Operations - Twist of Fate - scg41ea

### PR DESCRIPTION
**Pull request by YAmismo**
I'm back with another map, a GDI one this time.
For the map.png image, i have taken the one from the cnc64 map, is that ok? I don't know if any permission or credit is needed. I just guess is better to have the same image for all non campaign maps in the game.

### CODE REVIEW

File: mods/cnc/maps/gdi41/gdi41.lua
Comment (JovialFeline) [inline mods/cnc/maps/gdi41/gdi41.lua]: Dead buildings should already lack repair properties, or that's what I've gleaned from my script fiddling, anyway.

File: mods/cnc/maps/gdi41/gdi41.lua
Comment (JovialFeline) [inline mods/cnc/maps/gdi41/gdi41.lua]: Was this a mixed group at some point? Commando + Engineers would be my guess.

File: mods/cnc/maps/gdi41/gdi41.lua
Comment (JovialFeline) [inline mods/cnc/maps/gdi41/gdi41.lua]: It's possible to capture the temple before the nuke countdown (or yard trap) kicks in. If so, the player can get nuked by their newly captured temple. I'll admit this was funny to see happen.
```suggestion
	Trigger.OnKilledOrCaptured(NodTmpl, function()
		NukeDelay = 0
	end)
```
An owner check in `LaunchNuke` should accomplish the same thing if that's easier to read.

File: mods/cnc/languages/lua/en.ftl
Comment (JovialFeline) [inline mods/cnc/languages/lua/en.ftl]: I think some brevity would help, though I'm not fussy about how you choose to do it.
```suggestion
air-strikes-intel-report = Nod airstrikes are being directed by a communications center, located northwest. We may gain useful information from its capture.
```

File: mods/cnc/languages/lua/en.ftl
Comment (JovialFeline) [inline mods/cnc/languages/lua/en.ftl]: A similar thing here. It could be trimmed like this:
`Our engineers located Nod SAM sites. They also disabled an unusual trap or alarm on a construction yard to the south.`
If you'd like something more terse:
`Nod SAM sites located. An odd trap or alarm on their main construction yard was disabled.`
Whatever works.

**Review by JovialFeline — COMMENTED** [review]
Hello again. I took a while, but I've played through a few times now. I'd still like to make time for looking at the code, even if I lack knowledge of OpenRA's design. I'm posting "early" so you know this has some attention.

My multi-tasking in RTS games needs improvement, so it's hard to say how much of my difficulty here came from the layout or something else. Anyway, stuff from my notes:

- The Commando's newly available, among other things. I think I neglected to check this in [Eviction Notice](https://github.com/OpenRA/OpenRA/pull/20660). I don't yet know how closely map ports should follow, anyway.
- The MCV now enters once the whole ambush team is cleared, rather than just near the northern road. I don't mind this change since the OpenRA armor is quicker to retreat and save, and I can double-tap Q to send everybody south. The AI likes to chase, which can also make the opening moments less hassle.
- Some tanks and Chem Warriors are placed near the northern base spot, which seems unfortunate. Did you feel that start made the map too easy? Perhaps you didn't want to tinker with changing attack routes, especially the stealth tank one.
- The tweaked stealth tank waves prodded at my lack of TD knowledge. I had to check how their cloak works since they would sneak by the roadside Mammoths. The original seems to have them simply hunt from the road, which I'll agree is less interesting.
- An artillery piece now sits by the southern base spot, but it's not a bother to Orcas.

I did stumble on one crash that seems to be triggered by early BaseDefenseTeam deaths. The exception log: 
```Fatal Lua Error: [string "-- Make copies of global functions to preve..."]:30: Uncaught CLR exception at Lua->CLR boundary: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.InvalidOperationException: Attempted to get trait from destroyed object (arty 259 (not in world))
   at OpenRA.TraitDictionary.CheckDestroyed(Actor actor) in C:\Users\JovialFeline\OpenRA\OpenRA.Game\TraitDictionary.cs:line 84
   at OpenRA.Mods.Common.Scripting.TriggerGlobal.GetScriptTriggers(Actor a) in C:\Users\JovialFeline\OpenRA\OpenRA.Mods.Common\Scripting\Global\TriggerGlobal.cs:line 30
   at OpenRA.Mods.Common.Scripting.TriggerGlobal.OnAllKilled(Actor[] actors, LuaFunction func) in C:\Users\JovialFeline\OpenRA\OpenRA.Mods.Common\Scripting\Global\TriggerGlobal.cs:line 111
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at OpenRA.Scripting.ScriptMemberWrapper.Invoke(LuaVararg args) in C:\Users\JovialFeline\OpenRA\OpenRA.Game\Scripting\ScriptMemberWrapper.cs:line 76
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Eluant.LuaRuntime.MethodWrapper.Invoke(Object[] parms)
   at Eluant.LuaRuntime.MakeManagedCall(IntPtr state, MethodWrapper wrapper)
   at OpenRA.Scripting.ScriptContext.FatalError(String message)
   at OpenRA.Mods.Common.Scripting.TriggerGlobal.<>c__DisplayClass2_0.<AfterDelay>g__DoCall|0()
   at OpenRA.World.Tick()
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager)
   at OpenRA.Game.LogicTick()
   at OpenRA.Game.Loop()
   at OpenRA.Game.Run()
   at OpenRA.Game.InitializeAndRun(String[] args)
   at OpenRA.Launcher.Program.Main(String[] args)
```
The timing for the crash changes by difficulty, and I'd guess that's because of production delays. On Easy/Hard, the crash comes at 2:16/1:01 if a defender is killed beforehand. It's unlikely others would encounter this but if you can find the right spot, tossing in `IsDead` still seems worthwhile.

**Comment by YAmismo** [comment]
Hi, i have almost no free time lately, sorry. Anyway i'm done with your suggested changes :), fixed the crash with BaseDefenseTeam and a similar one with PatrolTeam (3 bikes at northwest)

> Some tanks and Chem Warriors are placed near the northern base spot, which seems unfortunate.

Well, i just thought this wasnt intended to be a base start (not sure why haha), and yes, the routes thing... should i better remove these units?

**Comment by JovialFeline** [comment]
> Hi, i have almost no free time lately, sorry. Anyway i'm done with your suggested changes :), fixed the crash with BaseDefenseTeam and a similar one with PatrolTeam (3 bikes at northwest)

No apologies necessary. I've mostly had my head down this week or so as well. Kudos for the changes.

> Well, i just thought this wasnt intended to be a base start (not sure why haha), and yes, the routes thing... should i better remove these units?

Oh, I'm sure the north side was **not** intended as a base start. I wouldn't have tried it in the original myself if I hadn't seen a video of a speedrunner building there, and it comes with some drawbacks. I was just surprised to see the guards there, figured you were aware of the spot, and had intended those guards to nudge the player south.

Removing the guards may suck as-is; stealth tanks lack stealth if the player's set up north, at least once the helipad outpost is gone. Some attack teams would probably be fine Hunting from the road to account for either base but I'd love to see the stealth behavior intact. It's very Nod. 😉

I'm content with the map's play, if it's not clear. I've had difficulty digging through the code for a review, and somebody with write access needs to look at this before a merge anyway, so I haven't been in a rush. I'll try to go through after this weekend, before I have to travel.

**Comment by YAmismo** [comment]
Removed some double spaces and renamed map files.

File: mods/cnc/maps/twist-of-fate/twist-of-fate.lua
Comment (JovialFeline) [inline mods/cnc/maps/twist-of-fate/twist-of-fate.lua]: A crash can occur here if a player doesn't first capture the airstrike HQ. This references an ID that's only made after the HQ's capture, so the capture is less optional than I'd thought.
```suggestion
		if DestroySams then
			Player1.MarkCompletedObjective(DestroySams)
		end
```

**Review by JovialFeline — CHANGES_REQUESTED** [review]
I made some more time for poking through the code, and I did find a potential show stopper. There could probably be style/convention changes too but being a newbie myself, I'm not comfy making recommendations beyond what I've already done.

Anyway, about that crash...

**Comment by YAmismo** [comment]
Fixed destroy sams objective crash, lol how i didn't see it.

**Comment by YAmismo** [comment]
Oops, i messed up all, xdd. Can i reopen the pr or must create it again?

**Pull request closed without merging**
